src/main.rs

```diff
@@ -420,6 +420,13 @@ impl SpiDrv {
         //     self.send_param(uart, param, last_param);
         // });
     }
+
+    fn pad_to_multiple_of_4(&mut self, uart: &mut EnabledUart, mut cmd: u8) {
+        while cmd % 4 == 0 {
```

Comment:
I'd like to see cmd come in as an immutable parameter and then you can assign it locally to a mutable u8 variable. It would be confusing as a caller of this function to see cmd be mutable and I'd wonder if there'd be side effects after it returns.

Reply:
Sure. I wrote that from conclusion that cmd is being passed by value and not reference so local variable would be redundant. However your approach might be more explicit

Reply:
I think that is still true what you said, it is transfering ownership. It's actually something for us both to review - what does passing a mutable parameter not by reference precisely mean in Rust?